.github/workflows/cache_dvc.yaml

```diff
@@ -0,0 +1,59 @@
+# Cache baseline images tracked by DVC and upload as artifacts.
+#
+# This workflow downloads dvc cache needed by PyGMT tests and uploads them as workflow
+# artifacts, which can then be accessed by other GitHub Actions workflows.
+#
+# This workflow serves as a workaround for the DagsHub authentication issue that
+# prevents PRs from forks to access the DVC remote.
+# Related issue: https://github.com/GenericMappingTools/pygmt/issues/4147
+#
+# It is scheduled to run every Sunday at 12:00 (UTC) and on the main branch if there are
+# any changes to the baseline images.
+#
+name: Cache DVC
+
+on:
+  push:
+    branches: [ main ]
+    paths:
+      - 'pygmt/tests/baseline/*.png.dvc'
+  pull_request:  # For testing only.
+  workflow_dispatch:
+  # Schedule runs on 12 noon every Sunday
+  schedule:
+    - cron: '0 12 * * 0'
+
+permissions: {}
+
+jobs:
+  dvc_cache:
```

Comment:
Could probably this as a separate job under `.github/workflows/cache_data.yaml` if want to reduce the number of workflow files.

Reply:
Ah yes, ok to keep them separate then.